### PR TITLE
Call Microsoft.Crank.JobObjectWrapper.exe with full path 

### PR DESCRIPTION
**Pull request by LoopedBard3**
Instead of relying on starting in the correct working directory, specify that we want the Microsoft.Crank.JobObjectWrapper.exe in the same location of the assembly/tool.

### CODE REVIEW

**Comment by sebastienros** [comment]
Can you add a functional test that will set cpuset to `0` just to trigger this code path? By copying exactly one of the existing one?

**Comment by sebastienros** [comment]
Example of one that can be used" https://github.com/dotnet/crank/blob/main/test/Microsoft.Crank.IntegrationTests/CommonTests.cs#L56

File: src/Microsoft.Crank.JobOjectWrapper/Program.cs [hunk 1]
```diff
@@ -30,8 +30,26 @@
 Console.WriteLine($"Filename: {process.StartInfo.FileName}");
 Console.WriteLine($"Args: {process.StartInfo.Arguments}");
 
+process.OutputDataReceived += (_, e) =>
```
Comment (sebastienros) [inline src/Microsoft.Crank.JobOjectWrapper/Program.cs:33]: I think it's equivalent to not invoking these:
```
RedirectStandardOutput = true,
RedirectStandardError = true,
```

Comment (LoopedBard3) [reply src/Microsoft.Crank.JobOjectWrapper/Program.cs:33]: These are both already being set in the StartInfo of the process starting this wrapper and the process being started by this wrapper.

**Comment by LoopedBard3** [comment]
> Can you add a functional test that will set cpuset to `0` just to trigger this code path? By copying exactly one of the existing one?

I added a test that uses --application.cpuSet 0 to test the code path. It would probably be good to also add one for a cpuSet range and one for multiple cpuSet ranges. Although I don't know the processor count of the normal test agents.

File: src/Microsoft.Crank.JobOjectWrapper/Program.cs [hunk 2]
```diff
@@ -30,8 +30,26 @@
...
+    }
+};
+
 process.Start();
 Console.WriteLine($"##ChildProcessId:{process.Id}");
+process.BeginOutputReadLine();
+process.BeginErrorReadLine();
 process.WaitForExit();
 
 Console.WriteLine(process.StandardOutput.ReadToEnd());
```
Comment (LoopedBard3) [inline src/Microsoft.Crank.JobOjectWrapper/Program.cs:55]: Is this line still necessary with this setup?

File: src/Microsoft.Crank.JobOjectWrapper/Program.cs (hunk 1 above)
Comment (LoopedBard3) [reply src/Microsoft.Crank.JobOjectWrapper/Program.cs:33]: As far as I can tell, just having the redirect enabled does not cause the parent process to actually receive the output live, and we need the output live in order to tell when the application has started.

Comment (sebastienros) [reply src/Microsoft.Crank.JobOjectWrapper/Program.cs:33]: > As far as I can tell, just having the redirect enabled does not cause the parent process to actually receive the output live

Was wondering the opposite, remove the code completely.

**Review by sebastienros — APPROVED** [review]

File: src/Microsoft.Crank.JobOjectWrapper/Program.cs (hunk 1 above)
Comment (LoopedBard3) [reply src/Microsoft.Crank.JobOjectWrapper/Program.cs]: Looks like that works as well, went ahead and updated to remove code rather than add code.

**Pull request merged**